src/Analyser/MutatingScope.php

```diff
@@ -3500,6 +3500,11 @@ private function shouldInvalidateExpression(string $exprStringToInvalidate, Expr
 			return false;
 		}
 
+		// Variables will not contain traversable expression. skip the NodeFinder overhead
+		if ($expr instanceof Variable) {
```

Comment:
Only if the variable name is string

Reply:
good catch. fixed and re-measured. same characteristics.. seems most variables contain strings :)